wicket/src/widgets/liveness.rs

```diff
@@ -0,0 +1,89 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::time::Duration;
+
+use tui::{
+    style::Style,
+    text::{Span, Spans},
+};
+
+use crate::defaults::colors::{OX_GRAY_DARK, OX_GREEN_LIGHT, OX_YELLOW};
+
+/// Tracker used by a single instance of liveness.
+#[derive(Debug)]
+pub struct LivenessState {
+    // Some means that the stopwatch hasn't yet been initialized.
+    stopwatch: Option<libsw::Stopwatch>,
+    live_threshold: Duration,
+}
+
+impl LivenessState {
+    pub fn new(live_threshold: Duration) -> Self {
+        Self { stopwatch: None, live_threshold }
+    }
+
+    /// Resets or initializes the stopwatch state.
+    pub fn reset(&mut self, elapsed: Duration) {
+        self.stopwatch = Some(libsw::Stopwatch::with_elapsed_started(elapsed));
+    }
+
+    /// Compute the liveness for this state.
+    pub fn compute(&self) -> ComputedLiveness {
+        if let Some(stopwatch) = &self.stopwatch {
+            let elapsed = stopwatch.elapsed();
+            if elapsed > self.live_threshold {
+                ComputedLiveness::Delayed(elapsed.as_secs())
+            } else {
+                ComputedLiveness::Live(elapsed.as_secs())
+            }
+        } else {
+            ComputedLiveness::NoResponse
+        }
+    }
+}
+
+/// Liveness that's been computed so far.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum ComputedLiveness {
+    /// The data is the number of seconds.
+    Live(u64),
+
+    /// The data is the number of seconds.
+    Delayed(u64),
+
+    NoResponse,
+}
+
+impl ComputedLiveness {
+    pub fn to_spans(&self) -> Spans<'static> {
+        match self {
+            ComputedLiveness::Live(secs) => Spans::from(vec![
+                Span::styled("live", Style::default().fg(OX_GREEN_LIGHT)),
```

Comment:
The style should probably be configurable through the `StatusBar` widget and passed in via`LivenessState`. This allows easy customization of individual screens. I went through quite a bit of iteration during initial development, such as changing colors of the rack: https://github.com/oxidecomputer/omicron/blob/main/wicket/src/screens/rack.rs#L135-L153

Reply:
Done, added a `LivenessStyles` stylesheet.